### PR TITLE
More player callbacks

### PR DESCRIPTION
**Pull request by JaceCear**

### CODE REVIEW

File: include/zones.h [hunk 1]
```diff
@@ -41,4 +41,10 @@
 
 #define CHAOS_EMERALDS_COMPLETED CHAOS_EMERALD(7)
 
+#define IsBossStage(lvl)                                                                \
+    ((((lvl)&0x3) == ACT_BOSS)                                                          \
```
Comment (freshollie) [inline include/zones.h:45]: `& ACTS_PER_ZONE`?

File: src/game/player_callbacks_1.c [hunk 2]
```diff
@@ -302,4 +325,491 @@ void PlayerCB_8025854(Player *player)
...
+void PlayerCB_8025AB8(Player *player)
+{
+    if ((player->moveState & (MOVESTATE_8000000 | MOVESTATE_8 | MOVESTATE_IN_AIR))
+        == MOVESTATE_8000000) {
+        PlayerCB_80273D0(player);
+    } else if ((player->moveState & (MOVESTATE_800 | MOVESTATE_8 | MOVESTATE_IN_AIR))
+               == MOVESTATE_800) {
+        sub_802A360(player);
+    } else {
+        if (player->unk99) {
```
Comment (freshollie) [inline src/game/player_callbacks_1.c:368]: better to write `if (player->unk99 != 0)` as this is not a bool

File: src/game/player_callbacks_1.c [hunk 3]
```diff
@@ -302,4 +325,491 @@ void PlayerCB_8025854(Player *player)
...
+
+        if ((player->speedGroundX > -Q_24_8(0.5))
+            && (player->speedGroundX < Q_24_8(0.5))) {
+            player->unk50 = 0;
+            player->speedGroundX = 0;
+        }
+
+        if (player->speedGroundX == 0) {
+            gPlayer.callback = PlayerCB_8025318;
+            gPlayer.callback(player);
```
Comment (freshollie) [inline src/game/player_callbacks_1.c:418]: Does this match with `PlayerCB_8025318(player)`? just wondering

File: src/game/player_callbacks_1.c [hunk 4]
```diff
@@ -302,4 +325,491 @@ void PlayerCB_8025854(Player *player)
...
+                if (player->moveState & MOVESTATE_40) {
+                    player->speedAirY += Q_24_8(12.0 / 256.0);
+                } else {
+                    player->speedAirY += Q_24_8(42.0 / 256.0);
+                }
+            }
+
+            player->x += player->speedAirX;
+
+            if ((gUnknown_03005424 ^ gUnknown_0300544C)
```
Comment (freshollie) [inline src/game/player_callbacks_1.c:453]: Probably need to make a macro for this

**Review by freshollie — APPROVED** [review]
Really great! Good work discovering more fields on player and naming rotation :D

File: src/game/player_callbacks_1.c (hunk 4 above)
Comment (freshollie) [reply src/game/player_callbacks_1.c:453]: Don't have to do now though

File: src/game/player_callbacks_1.c (hunk 3 above)
Comment (JaceCear) [reply src/game/player_callbacks_1.c:418]: Huh, it actually does.
I only tried player->callback(player) before, which didn't work.

Guess it makes more sense to write it this way?

File: src/game/player_callbacks_1.c (hunk 4 above)
Comment (JaceCear) [reply src/game/player_callbacks_1.c:453]: Once we know WTF these variables are, sure! :D

**Pull request merged**